src/lib.rs

```diff
@@ -1,2 +1,396 @@
 extern crate libc;
-include!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/arrayfire.rs"));
\ No newline at end of file
+
+use std::fmt;
+use libc::c_void;
+use libc::c_int;
+use libc::c_uint;
+use libc::c_double;
+use libc::c_longlong;
+
+use std::ops::Index;
+use std::ops::Add;
+
+#[link(name="afcpu")]
```

Comment:
@jramapuram that's interesting, how would it know which library to pick the symbol from ? or is rustc clever enough to search all the libraries it is linking with to find the requested symbol ?